### PR TITLE
store: getStructFields now take pointers

### PR DESCRIPTION
**Pull request by chipaca**
`getStructFields` now takes a pointer to the struct.
So you can `getStructFields((*myThing)(nil))`...

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -270,8 +271,8 @@ func respToError(resp *http.Response, msg string) error {
 	return fmt.Errorf(tpl, msg, resp.StatusCode, resp.Request.Method, resp.Request.URL)
 }
 
-func getStructFields(s interface{}) []string {
-	st := reflect.TypeOf(s)
+func getStructFields(s interface{}, exceptions ...string) []string {
```
Comment (mvo5) [inline store/store.go]: API wise, I wonder if we should have a `getStructFields` and `getStructFieldsWithExceptions` ? Looking at the usage of `getStructFields` its not clear what the second arg means (without looking at the definition).

File: store/store.go [hunk 2]
```diff
@@ -420,11 +421,8 @@ type sectionResults struct {
 	} `json:"_embedded"`
 }
 
-// The fields we are interested in
-var detailFields = getStructFields(snapDetails{})
-
-// The fields we are interested in for snap.ChannelSnapInfos
-var channelSnapInfoFields = getStructFields(channelSnapInfoDetails{})
+// The fields we are interested in (not you, snap_yaml_raw)
+var detailFields = getStructFields((*snapDetails)(nil), "snap_yaml_raw")
```
Comment (mvo5) [inline store/store.go]: This is the motivation for the PR, right? Might be nice to put that into the PR description *why* this was changed from a struct to a pointer. Actually, is that really the reason? Or could we have the old way of looking at the struct just with exceptions?

File: store/store.go [hunk 3]
```diff
@@ -506,6 +504,16 @@ func (s *Store) defaultSnapQuery() url.Values {
 	return q
 }
 
+func (s *Store) queryForSnapInfo() url.Values {
```
Comment (mvo5) [inline store/store.go]: I think would change the PR description a bit, the PR makes it sound like its about getStructFields refactor but its about more functional stuff it seems :)

Comment (chipaca) [reply store/store.go]: I think I failed to mention this PR was stacked on #4748...

**Comment by mvo5** [comment]
Hm, most of my questions are still there :/ 1. what is the advantage of the pointer over the value? 2. would it make sense to have "getStructFields" and "getStructFieldsExcept" to make the API nicer?

**Comment by chipaca** [comment]
Hm, I thought I'd replied to the ponter-vs-struct one somewhere, but can't find it now.
The advantage of passing the pointer (which is really just a typed `nil`) is that you don't need to pass around a huge struct.
I don't think it makes sense to have two functions, as it's only called in one place.
This last point really makes the pointer-vs-struct thing silly (and I don't mind if this pr bounces), but it was fun to do.

**Comment by mvo5** [comment]
thanks @chipaca !

**Pull request merged**